service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -449,7 +449,20 @@ func (d *Decoder) decodeMap(avMap map[string]*dynamodb.AttributeValue, v reflect
 		fields := unionStructFields(v.Type(), d.MarshalOptions)
 		for k, av := range avMap {
 			if f, ok := fieldByName(fields, k); ok {
-				fv := v.FieldByIndex(f.Index)
+				fv := v
+				for i, x := range f.Index {
```

Comment:
The section in encode and decode look very similar it would be helpful to reduce the duplicated logic between the two.